cpp/benchmarks/io/cuio_common.cpp

```diff
@@ -141,3 +141,16 @@ std::vector<cudf::size_type> segments_in_chunk(int num_segments, int num_chunks,
 
   return selected_segments;
 }
+
+void drop_cache()
+{
+  if (std::getenv("CUDF_BENCHMARK_DROP_CACHE") != nullptr) {
+    std::array drop_cache_cmds{"/sbin/sysctl vm.drop_caches=3",
+                               "sudo /sbin/sysctl vm.drop_caches=3"};
+
+    CUDF_EXPECTS(std::any_of(drop_cache_cmds.cbegin(),
+                             drop_cache_cmds.cend(),
+                             [](auto& cmd) { return std::system(cmd) == 0; }),
```

Comment:
Pardon my ignorance, but I'm not familiar with `std::system()` and its return value. `0` indicates failure, and not success?

Reply:
The [`cppreference` docs](https://en.cppreference.com/w/cpp/utility/program/system) indicate `std::system()` *usually* returns what the invoked program returns.

Testing on an Ubuntu install, both commands seem to have an exit status of `0`. :/ I'm pretty sure I haven't grokked this.

Reply:
> std::system() usually returns what the invoked program returns. 

My read of this is that it returns zero on success. 

For additional clarification, this code tries different options to clear the cache until one works. We assume that `any_of` returns once the predicate returns true.

Reply:
Two additional considerations:
1. The suggestion seems to be to use [WEXITSTATUS + WSTOPSIG](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wait.html) to interpret the return value. That suggestion might just be a guideline.
2. [The docs](https://en.cppreference.com/w/cpp/utility/program/system), indicate that an explicit flush of `std::cout` is necessary before calling `std::system()`. Since this is a benchmark, I'm guessing there is screen I/O.

I think maybe we can ignore #1. But we might consider the `std::flush`.

Reply:
Whoops sorry I think there were additional responses posted here while I was typing mine out. I don't think there is a _guarantee_ that a command return code of zero will be translated to a returned value of zero. Based on [the description of the macros](https://pubs.opengroup.org/onlinepubs/9699919799/functions/wait.html), `WEXITSTATUS` is pulling out the bottom eight bits of the number, which means that the other 24 bits of the int could vary freely between different implementations and still remain POSIX-compliant. Whether that happens in practice, I don't know. Maybe something @robertmaynard has more insight into?

Reply:
> this code tries different options to clear the cache until one works.

Here is my attempt at dropping caches on an Ubuntu 20.04 machine, without root:
```
$ /sbin/sysctl vm.drop_caches=3
sysctl: permission denied on key "vm.drop_caches", ignoring
$ echo $?
0
```
My fear is if the command doesn't return `0` on failure to drop caches.

Reply:
>  I think there were additional responses posted here while I was typing mine out. 

Argh. Likewise.

Reply:
It looks like the issue is that `sysctl` doesn't return an error code when asked to update keys that you don't have permission too.

So in all cases the return value can't be considered a trusted source of success for the request.

We would need to cat / read some output to determine if the command(s) executed

Reply:
> It looks like the issue is that sysctl doesn't return an error code when asked to update keys that you don't have permission too.

:/ Regrettably, I can confirm on Ubunutu 20.04.
```c++
  std::cout << "Attempting /sbin/sysctl vm.drop_caches=3: " << std::endl;
  auto ret = std::system("/sbin/sysctl vm.drop_caches=3");
  std::cout << "Return code ret == " << ret << std::endl;
  std::cout << "WEXITSTATUS(ret)== " << WEXITSTATUS(ret) << std::endl;
  std::cout << "WSTOPSIG(ret)   == " << WSTOPSIG(ret) << std::endl;
```
Output:
```
Attempting /sbin/sysctl vm.drop_caches=3:
sysctl: permission denied on key "vm.drop_caches", ignoring
Return code ret == 0
WEXITSTATUS(ret)== 0
WSTOPSIG(ret)   == 0
```

Reply:
What rotten luck. Working on capturing the output to check for errors.
Thanks everyone for digging into this!

Reply:
Thank you for your patience with this, @vuule. Resolving this conversation now.